src/errors/InvalidResourceError.ts

```diff
@@ -5,10 +5,19 @@
 
 // eslint-disable-next-line import/prefer-default-export
 export class InvalidResourceError extends Error {
+    readonly isInvalidResourceError: boolean;
+
     constructor(message = 'Invalid Resource') {
         // Node Error class requires passing a string message to the parent class
         super(message);
         Object.setPrototypeOf(this, InvalidResourceError.prototype);
+        this.isInvalidResourceError = true;
         this.name = this.constructor.name;
     }
 }
+export function isInvalidResourceError(err: InvalidResourceError | unknown): err is InvalidResourceError {
+    return (
+        (err as InvalidResourceError).isInvalidResourceError !== undefined &&
+        (err as InvalidResourceError).isInvalidResourceError === true
```

Comment:
this crashes if `err` is undefined or null.

Also, the `!== undefined` is redundant if we are already doing `=== true`